shopfloor_base/models/shopfloor_scenario.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 ACSONE SA/NV (http://www.acsone.eu)
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from odoo import api, fields, models
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+from odoo.addons.http_routing.models.ir_http import slugify
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import yaml
+except ImportError:
+    _logger.debug("`yaml` lib is missing")
+
+
+class ShopfloorScenario(models.Model):
+    _name = "shopfloor.scenario"
+    _description = "Shopfloor Scenario"
+
+    name = fields.Char(required=True, translate=True)
+    key = fields.Char(required=True)
```

Comment:
what is the purpose of key ? 
Not blocking but, can't we work without it ?

Reply:
it is mandatory: you really want 1 and only one scenario to match the usage of a component.

Reply:
Why not a db id then ?

Reply:
IDG your point. I could you do it via ID? This is something the integrator/dev should provide and it should match the usage on both back and front. It's exactly what was used before in `scenario` as string. 
I could change it to `usage` if you prefer the name but we need something like this anyway.

Reply:
I suggested id because I misunderstood the meaning of key.

Does it replace that : https://github.com/OCA/wms/pull/169/files/ebce44f5d58c96bc8cc208c3f3e87d2f32d0abcd#diff-6b10fc808919519f099c8197a6ee5aadca6444fb042593847fe1066d23228657L84 ?

Reply:
Yes, it replaces the string `scenario` that was on the menu model.

Reply:
Personally, I prefer when this kind of fields are field Selection. From an UX perspective, I don't see why a user has to enter it itself. 

Because it's really a minor point, it's ok to leave it as is for the moment.

Reply:
well, in theory the user is not the one that should edit this. I don't expect anyone to create scenario via UI as it would be useless. So, on the next round we should probably:

* make it not editable if not w/ very high permission
* make it a select